service/history/replication/executable_history_task.go

```diff
@@ -299,3 +305,134 @@ func (e *ExecutableHistoryTask) getDeserializedEvents() (_ []*historypb.HistoryE
 	}
 	return events, newRunEvents, err
 }
+
+func (e *ExecutableHistoryTask) BatchWith(incomingTask BatchableTask) (TrackableExecutableTask, bool) {
+	if !e.wantToBatch {
```

Comment:
locking?

Reply:
Same as `batchStateOpen` in above, `wantToBatch` is also a one way switch.